beacon-chain/core/helpers/committee.go

```diff
@@ -440,14 +439,30 @@ func ProposerAssignments(
 		if slot == 0 {
 			continue
 		}
+
 		if err := state.SetSlot(slot); err != nil {
 			return nil, err
 		}
+
 		i, err := BeaconProposerIndex(state)
 		if err != nil {
 			return nil, errors.Wrapf(err, "could not check proposer at slot %d", state.Slot())
 		}
-		proposerIndexToSlots[i] = append(proposerIndexToSlots[i], slot)
+
+		pubKey := state.PubkeyAtIndex(i)
+		proposerSlots := make([]types.Slot, 0)
+		proposerSlots = append(proposerSlots, slot)
+		assign := &ethpb.ValidatorAssignments_CommitteeAssignment{
+			ProposerSlots:  proposerSlots,
+			PublicKey:      pubKey[:],
+			ValidatorIndex: i,
+		}
+
+		proposerIndexToSlots = append(proposerIndexToSlots, assign)
+
+		if len(proposerIndexToSlots) > sliceRange {
```

Comment:
This should never happen. But I've manage to leave some condition for this and have clean proper log output to see what was the state (all variables etc.) when it occur.